torch/lib/c10d/ProcessGroupGloo.hpp

```diff
@@ -132,11 +132,15 @@ class ProcessGroupGloo : public ProcessGroup {
     int srcRank_;
   };
 
-  struct Options {
+  struct Options : public ProcessGroup::Options {
     explicit Options();
 
+    // return intrusive_ptr of the object
+    static c10::intrusive_ptr<Options> create() {
```

Comment:
Is there a reason we've converted to always returning intrusive_ptr of the options instead of the regular struct as before?

Reply:
Mostly for TorchScript compliance, custom classes are better supported in TorchScript if we pass them by intrusive_ptr.